### PR TITLE
[Feature] Signup Functionality

### PR DESCRIPTION
**Pull request by pladreyt**
## Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):
​
## What is the current behavior?

User can signup by filling **email**, **first name** and **last name**.

Issue Link: [Issue](https://trello.com/c/mBZn89Il/2-5-as-a-user-i-should-be-able-to-sign-up-so-my-account-is-created-in-target-mvd)
​
## What is the new behavior?
User can signup filling the previous fields and **gender**.
​
## Other information

N/A
​
## Preview
N/A

### CODE REVIEW

File: src/database/factories/user.factory.ts [hunk 1]
```diff
@@ -5,12 +5,14 @@ import { User } from '@entities/user.entity';
 define(User, (faker: typeof Faker) => {
   const firstName = faker.name.firstName();
   const lastName = faker.name.lastName();
+  const gender = faker.random.arrayElement(['male', 'female', 'nonbinary']);
```
Comment (EMazzaglia) [inline src/database/factories/user.factory.ts]: What do you think about using an Enum to store these values. So you can avoid breaking the DRY principle.

File: src/dto/signUpDTO.ts [hunk 2]
```diff
@@ -3,8 +3,11 @@ import { BaseUserDTO } from './baseUserDTO';
 
 export class SignUpDTO extends BaseUserDTO {
   @IsString()
-  firstName?: string;
+  firstName!: string;
```
Comment (EMazzaglia) [inline src/dto/signUpDTO.ts:7]: Are we sure that @isString() is not allowing nulls? We can check the docs https://github.com/typestack/class-validator

Comment (EMazzaglia) [reply src/dto/signUpDTO.ts:7]: Same applies for `gender` and `lastName`

File: src/entities/user.entity.ts [hunk 3]
```diff
@@ -2,11 +2,14 @@ import { Column, Entity, Index } from 'typeorm';
...
-  firstName?: string;
+  @Column()
+  firstName!: string;
+
+  @Column()
+  lastName!: string;
 
   @Column({ nullable: true })
-  lastName?: string;
+  gender?: string;
```
Comment (EMazzaglia) [inline src/entities/user.entity.ts]: This is a required field, we should update this!

File: __tests__/controllers/users/update-user.spec.ts [hunk 4]
```diff
@@ -6,7 +6,7 @@ import { User } from '@entities/user.entity';
 import { API } from '../../utils';
```
Comment (EMazzaglia) [inline __tests__/controllers/users/update-user.spec.ts:6]: Make sure that we are pointing to the dev DB and not production. WE dont want new fake users (?

**Review by EMazzaglia — CHANGES_REQUESTED** [review]
There are some changes requested and research about some concerns on how we handle the mandatory fields of `gender` `male` etc etc

File: __tests__/controllers/users/update-user.spec.ts [hunk 5]
```diff
@@ -16,11 +16,14 @@ describe('updating a user', () => {
     const id = user.id;
     user.firstName = 'new firstname';
     user.lastName = 'new lastname';
+    user.gender = 'new_gender';
```
Comment (EMazzaglia) [inline __tests__/controllers/users/update-user.spec.ts]: We should create a function that generates ready to use mocked users. So we avoid breaking the DRY principle

File: src/entities/user.entity.ts (hunk 3 above)
Comment (pladreyt) [reply src/entities/user.entity.ts]: Thanks, updated in new commit.

File: src/dto/signUpDTO.ts (hunk 2 above)
Comment (pladreyt) [reply src/dto/signUpDTO.ts:7]: Added isNotEmpty in new commit

File: src/database/factories/user.factory.ts (hunk 1 above)
Comment (pladreyt) [reply src/database/factories/user.factory.ts]: Great suggestion, I updated it and did it with Enum in the new commit.

File: __tests__/controllers/users/update-user.spec.ts (hunk 4 above)
Comment (pladreyt) [reply __tests__/controllers/users/update-user.spec.ts:6]: Made sure it is pointing to the dev db.

**Review by EMazzaglia — APPROVED** [review]
LGTM, We should add unit tests for this functionality. 
Nice job!

**Pull request merged**